### ISSUE TITLE
firewall: Add services dialog should have primary button disabled if nothing is chosen

### ISSUE THREAD

**Issue by KKoukiou** [cockpit-project/cockpit#18419]
### Explain what happens

1. Go to firewall page and click on 'Add services' 
2.  Click 'Add services' primary button without selecting anything
3.  No validation is done and the dialog closes

Same happens to the 'Add port' version of the dialog. Even worse there, the firewall configuration if appended with some invalid entry. 

### Version of Cockpit

284

### Where is the problem in Cockpit?

Firewall

### Server operating system

Fedora

### Server operating system version

38

### What browsers are you using?

Firefox

**Comment by vaibhav-1508** [comment]
Hey there! I would like to work on this issue

**Comment by omarY23** [comment]
Hey @KKoukiou. Would adding an alert component on top with a message after clicking 'Add services' button without selecting anything work? Or should simply show another modal upon clicking?

**Comment by KKoukiou** [comment]
@omarY23 just disable the primary (blue) button if no service is chosen in the 'Services' view and similarly if no port is specified in the 'Ports' view.

**Comment by omarY23** [comment]
I am having issue in creating a PR. I forked this repo in my system and committed my changes still an authentication error while publishing my branch.
![Screenshot 2023-03-01 175623](https://user-images.githubusercontent.com/111054035/222139463-20d6314f-24d8-484c-ad7e-6e83bcfd0715.png)

**Comment by KKoukiou** [comment]
@omarY23 please don't use the issues for technical help. For this please join our IRC channel. https://github.com/cockpit-project/cockpit/wiki/GSoC-2023-topics#contact

This error is from GitHub desktop and has nothing to do with cockpit itself from what I can see. For more info use the IRC ^

**Comment by MahmoudHamdy02** [comment]
Hello! While I was working on this issue I found that if I select a service from "Services", and then click on "Custom ports", the service will still be selected. This might create some issues since both menus share the same "add" button.

This is an example of what happens when I add this condition to the button `isDisabled={!this.state.selected.size && !this.state.custom_tcp_value && !this.state.custom_udp_value}`
-> The button starts out disabled 
-> Select a service
-> Button is enabled
-> Switch to "Custom ports"
-> Button is still enabled, even if all port input fields are empty
-> Clicking button gives an error

Is this the intended behavior, or should the selections reset when switching menus? Thanks!

**Comment by KKoukiou** [comment]
I don't think we should reset the dialog. Probably it's enough to adjust the check for disabled to also check the mode.

isDisabled={(this.state.custom && (!this.state.custom_tcp_value && !this.state.custom_udp_value)) || (!this.state.custom && !this.state.selected.size)}

**Comment by kartikvirendrar** [comment]
Hello @KKoukiou, is this issue still open I would like to work on it

**Comment by MaheshKarhale2111** [comment]
@KKoukiou I would like to work on this issue. Please let me know if the issue is still open or not

**Comment by abhishekkhanna2710** [comment]
I want to work on this issue ? Can anyone assigned me the task for that

**Comment by KKoukiou** [comment]
> I want to work on this issue ? Can anyone assigned me the task for that

It's already taken.
